pkg/api/controller.go

```diff
@@ -2107,6 +2107,150 @@ func (c *Controller) ResetBranch(w http.ResponseWriter, r *http.Request, body Re
 	writeResponse(w, r, http.StatusNoContent, nil)
 }
 
+func (c *Controller) ImportStart(w http.ResponseWriter, r *http.Request, body ImportStartJSONRequestBody, repository, branch string) {
+	if !c.authorize(w, r, permissions.Node{
+		Type: permissions.NodeTypeAnd,
+		Nodes: []permissions.Node{
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.ImportFromStorage,
+					Resource: permissions.StorageNamespace("*"),
+				},
+			},
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.WriteObjectAction,
+					Resource: permissions.BranchArn(repository, branch),
+				},
+			},
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.CreateBranchAction,
+					Resource: permissions.RepoArn(repository),
+				},
+			},
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.CreateCommitAction,
+					Resource: permissions.BranchArn(repository, branch),
+				},
+			},
+		},
+	}) {
+		return
+	}
+
+	ctx := r.Context()
+	c.LogAction(ctx, "import", r, repository, branch, "")
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, "missing user")
+		return
+	}
+	var metadata map[string]string
+	if body.Commit.Metadata != nil {
+		metadata = body.Commit.Metadata.AdditionalProperties
+	}
+	paths := make([]catalog.ImportPath, 0, len(body.Paths))
+	for _, p := range body.Paths {
+		pathType, err := catalog.GetImportPathType(p.Type)
+		if c.handleAPIError(ctx, w, r, err) {
+			return
+		}
+		paths = append(paths, catalog.ImportPath{
+			Destination: p.Destination,
+			Path:        p.Path,
+			Type:        pathType,
+		})
+	}
+
+	committer := user.Username
+	importID, err := c.Catalog.Import(r.Context(), repository, branch, catalog.ImportRequest{
```

Comment:
I agree - but we should first understand what permissions are required for querying import status. I want to think that if a user has read permissions to the target branch - then it could query for status regardless of who created it